src/test/java/org/alfresco/repo/policy/PolicyComponentTransactionTest.java

```diff
@@ -431,6 +443,31 @@ public void behaviourHierarchyTestWork(QName createDocType, ClassFilter... disab
         }
     }
 
+    public void behaviourHierarchyNodeTestWork(QName createDocType, QName... disableTypes) throws Exception
+    {
+        UserTransaction transaction = trxService.getUserTransaction();
+        try
+        {
+            transaction.begin();
+            NodeRef nodeRef = createDocOfType(createDocType);
+            disableBehaviours(nodeRef, disableTypes);
+            try
+            {
+                nodeService.addAspect(nodeRef, ASPECT, null);
+            }
+            finally
+            {
+                enableBehaviours(nodeRef, disableTypes);
+            }
+            transaction.commit();
+        }
+        catch(Exception e)
+        {
+            try { transaction.rollback(); } catch (IllegalStateException ee) {}

```

Comment:
This try catch block shouldn't be on a single line. I'll merge it and format the code in the following commit.